packages/zero-tag-processor/src/keyframes.ts

```diff
@@ -0,0 +1,154 @@
+import type { Expression } from '@babel/types';
+import { validateParams } from '@linaria/tags';
+import type {
+  CallParam,
+  TemplateParam,
+  Params,
+  TailProcessorParams,
+  ValueCache,
+} from '@linaria/tags';
+import type { Replacements, Rules } from '@linaria/utils';
+import { ValueType } from '@linaria/utils';
+import type { CSSInterpolation } from '@emotion/css';
+import BaseProcessor from './base-processor';
+import type { IOptions } from './styled';
+import { cache, keyframes } from './utils/emotion';
+
+type Primitive = string | number | boolean | null | undefined;
+
+export default class KeyframesProcessor extends BaseProcessor {
+  callParam: CallParam | TemplateParam;
+
+  constructor(params: Params, ...args: TailProcessorParams) {
+    super(params, ...args);
+    if (params.length < 2) {
+      throw BaseProcessor.SKIP;
+    }
+    validateParams(
+      params,
+      ['callee', ['call', 'template']],
+      `Invalid use of ${this.tagSource.imported} tag.`,
+    );
+
+    const [, callParams] = params;
+    if (callParams[0] === 'call') {
```

Comment:
As far as I understand this code it accepts functions and string templates, but we need to support JavaScript objects as well.

Reply:
We should also add an example for this, in order to validate that it works as expected.

Reply:
The example is [here](https://github.com/mui/material-ui/pull/39155/files#diff-f7e1a22a188de1696765d207773909b2493fefe850c11c5917194a212b30074aR23). Function or Object has same notion for Linaria. So the check here is whether its tagged template or function/object.

Reply:
The logic in the code checks that if it is a function call, ie, `keyframes()` and in `else`, it checks if it is called like this

```
keyframes`
`
```